IntegrationWithPharmacies/Controllers/SharingPrescriptionController.cs

```diff
@@ -1,30 +1,198 @@
-
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Mail;
+using System.Text;
+using HealthClinic.CL.DbContextModel;
+using HealthClinic.CL.Service;
+using IntegrationWithPharmacies.FileProtocol;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging.Abstractions;
+using RestSharp;
+
+namespace IntegrationWithPharmacies.Controllers
+{
+
+    [Route("api/[controller]")]
+    [ApiController]
+    public class SharingPrescriptionController : Controller
+    {
+        private String Environment { get; set; }
+        private MedicineService MedicineService { get; set; }
+        private PatientService PatientService { get; set; }
+        public SharingPrescriptionController(MyDbContext context)
+        {
+            MedicineService = new MedicineService(context);
+            PatientService = new PatientService(context);
+            Environment = "Development"; 
+        }
+
+        [HttpGet("patients")]
+        public IActionResult GetPatients()
+        {
+            return Ok(PatientService.GetAll());
+        }
+        [HttpGet]
+        public IActionResult Get()
+        {
+            return Ok(MedicineService.GetAll());
+        }
+
+        [HttpGet("medicinesIsa")]
+        public IActionResult GetMedicinesFromIsa()
+        {
+            var client = new RestSharp.RestClient("http://localhost:8082");
+            var request = new RestRequest("/medicineRequested");
+            var response = client.Get<List<MedicineName>>(request);
+            Console.WriteLine("Status: " + response.StatusCode.ToString());
+            List<MedicineName> result = response.Data;
+            result.ForEach(medicine => Console.WriteLine(medicine.ToString()));
+            return Ok(result);
+        }
+
+        [HttpPost]
+        public IActionResult Post(Prescription prescription)
+        {
+            if (Environment.Equals("Local")) return sendPrescriptionSftp(prescription);
+            return BadRequest();
+        }
+
+        private IActionResult sendPrescriptionSftp(Prescription prescription)
+        {
+            var sftpService = new SftpService(new NullLogger<SftpService>(), getConfig());
+            String complete = @"FilePrescriptions\..\Prescription" + DateTime.Now.ToString("dd-MM-yyyy") + "_" + getRandomNumber() + ".txt";
+            System.IO.FileStream fs = System.IO.File.Create(complete);
+            fs.Close();
+            System.IO.File.WriteAllText(complete, getTextForPrescription(prescription));
+            sftpService.UploadFile(@"..\test.txt", @"\pub\" + complete);
+            SendNotificationAboutReport();
+            return Ok();
+        }
+
+        [HttpGet("http/recieve/{medicine}")]
+        public IActionResult GetMedicineDescription(string medicine)
+        {
+            var url = "http://localhost:8082/upload/medicine/" + medicine;
+            HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create(url);

```

Comment:
Write this in form:
"HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create("http://localhost:8082/upload/medicine/" + medicine"http://localhost:8082/upload/medicine/" + medicine);"